In [3]:
from pyproj import Transformer

#### Hypocentres

In [4]:
transformer = Transformer.from_crs("epsg:32627", "epsg:4326", always_xy=True)
#transformer = Transformer.from_crs("utm27", "epsg:4326", always_xy=True)

|                	| x           	| y            	| z     	|
|----------------	|-------------	|--------------	|-------	|
| Simple East    	| 670950.2    	| 7327297.5    	| -7000 	|
| Simple Middle  	| 639613.9    	| 7340258.5    	| -7000 	|
| Simple West    	| 606405.1    	| 7357136.2    	| -7000 	|
| Complex East   	| 663105.107  	| 7331567.26   	| -7000 	|
| Complex Middle 	| 642704.0    	| 7339196.2    	| -7000 	|
| Complex West   	| 607077.3925 	| 7349647.4498 	| -7000 	|

In [5]:
x = [670950.2, 639613.9, 606405.1, 663105.107, 642704.0, 607077.3925]
y = [7327297.5, 7340258.5, 7357136.2, 7331567.26, 7339196.2, 7349647.4498]

In [6]:
# project back to geocentric coordinates
lon, lat = transformer.transform(x, y)

In [7]:
print(lon)

[-17.22866619559341, -17.904499706910716, -18.62572578181979, -17.395899999654276, -17.837300906465217, -18.617099999404182]


In [8]:
print(lat)

[66.02013608073102, 66.15159983086038, 66.3158548848616, 66.06250000797478, 66.14069969008978, 66.24850000855942]


|                	| lon      	| lat     	|
|----------------	|----------	|---------	|
| Simple East    	| -17.2286 	| 66.0201 	|
| Simple Middle  	| -17.9044 	| 66.1515 	|
| Simple West    	| -18.6257 	| 66.3158 	|
| Complex East   	| -17.3958 	| 66.0625 	|
| Complex Middle 	| -17.8373 	| 66.1406 	|
| Complex West   	| -18.6170 	| 66.2485 	|

#### Modal domain

In [9]:
x_dom = [525000, 730000]
y_dom = [7.248e+06, 7.43e+06]

In [11]:
lon_dom, lat_dom = transformer.transform(x_dom, y_dom)
print(lon_dom, lat_dom)

[-20.462729265464844, -15.741424836251399] [65.35386850808597, 66.90045035328437]
